### ISSUE TITLE
NovoGrad with beta2=0 turns it into AdaGrad?

### ISSUE THREAD

**Issue by vadimkantorov** [NVIDIA/DeepLearningExamples#186]
In https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/SpeechRecognition/Jasper is stated that beta2=0 was used. Does it mean that this turns the algorithm into AdaGrad, and AdaGrad was used for producing the Jasper results?

Thanks!

**Comment by blisc** [comment]
I'm not too familiar with AdaGrad, but from a quick glance, here are some key differences between Novograd with beta2=0 and AdaGrad:

- Novograd utilizes momentum (first moment) which is controlled using beta1
- We decouple weight decay from the computation of the gradient g_t
- First and second moments are calculated per layer as opposed to per parameter

For further details, see the Novograd paper: https://arxiv.org/abs/1905.11286

**Comment by yzhang123** [comment]
First of all, Adadelta doesnt have a moving average, Novograd has. 
Were you thinking of beta1=0? If beta1=0, 1st moment average falls away. Then it would be similar to Adagrad, but not exactly the same since the square root normalization factors are different: Novograd computes layer-wise gradient normalization, whereas AdaGrad and Adam do parameter-wise (per element).

**Comment by vadimkantorov** [comment]
Thanks for your clarifications! I thought that it is similar to autograd because both use norm of the current gradients for normalizing the update.

After reading the NovoGrad paper it seemed that beta2>0 helped too, but it is may be not the case :)

Though the other details matter too. Thanks!
